contracts/true-currencies/TrueCurrencyWithPoR.sol

```diff
@@ -49,13 +49,11 @@ abstract contract TrueCurrencyWithPoR is TrueCurrency, IPoRToken {
         // Get required info about total supply & decimals
         uint8 trueDecimals = decimals();
         uint8 reserveDecimals = AggregatorV3Interface(chainReserveFeed).decimals();
+
+        // Decimals of the Proof of Reserve feed must be the same as the token's.
+        require(reserveDecimals == trueDecimals, "Unexpected decimals of PoR feed");
```

Comment:
Since we no longer need to use the `trueDecimals` and `reserveDecimals` local vars, I would recommend shortening this to a single line to reduce clutter and improve clarity (adding a local variable means auditors will search for all possible abuses of this variable).
```suggestion
        // Decimals of the Proof of Reserve feed must be the same as the token's.
        require(decimals() == AggregatorV3Interface(chainReserveFeed).decimals(), "Unexpected decimals of PoR feed");
```

Reply:
Actually, let's move this decimals check up above the `latestRoundData()` call to align with the [checks-effects-interactions](https://docs.soliditylang.org/en/develop/security-considerations.html?highlight=check%20effects#use-the-checks-effects-interactions-pattern) pattern. Now that we don't consume decimals here, variable proximity-to-use is no longer a concern.

(To expand on this a little more: `decimals()` is a cheaper check than calling `latestRoundData()` to get a struct. Also, unlike `latestRoundData()`, `decimals()` should never change between invocations, so in some sense having a contract with the right decimals is a precondition for having correct round data from that contract.)

Reply:
Thanks for the first fix (inlining)!

Could you also move this require and its comments up above the `latestRoundData()` call? The order shouldn't affect the conditions for when this function call would revert, but it's a more defensive code style and improves clarity:
- aggregator decimals should never change, so we want to fail earlier if this ever happens
- aggregator latestRoundData() often changes, so only _some_ calls to it should revert

Concretely, if we're in a situation where _both_ the updatedAt heartbeat is stale _and_ decimals are wrong, then we should revert with the reason that decimals are wrong, because it indicates a more serious problem that needs to be addressed. Thanks!

Reply:
Done, sorry I forgot about the reordering earlier.